portal-kernel/src/com/liferay/portal/kernel/util/HttpComponentsUtil.java

```diff
@@ -252,7 +252,9 @@ public static String getDomain(String url) {
 			uri = getURI(url);
 		}
 		catch (URISyntaxException uriSyntaxException) {
-			_log.error(uriSyntaxException);
+			if (_log.isDebugEnabled()) {
```

Comment:
@JanisZhang The original `HttpImpl` has the try-catch-log-debug block in the `getURI(String)` method. When we moved this logic, the method was changed to directly throw the exception. This causes logic changes not only in the methods/logic modified in this PR.

For example, in `LayoutReferenceExportImportContentProcessor#_getPortalURL`,

```
	private String _getPortalURL(String url, String portalURL)
		throws PortalException {

		try {
			URI uri = HttpComponentsUtil.getURI(url);

			if ((uri != null) &&
				InetAddressUtil.isLocalInetAddress(
					InetAddress.getByName(uri.getHost()))) {

				return StringBundler.concat(
					uri.getScheme(), "://", uri.getHost(), StringPool.COLON,
					uri.getPort());
			}
		}
		catch (UnknownHostException unknownHostException) {
			if (_log.isDebugEnabled()) {
				_log.debug(unknownHostException);
			}
			else if (_log.isWarnEnabled()) {
				_log.warn(unknownHostException);
			}
		}
		catch (Exception exception) {
			throw new PortalException(exception);
		}

		return portalURL;
	}
```

The logic is to do a null check upon the return value, and return the original string if the `uri` is null. After the move, `getURI` throws an exception, so now for an illegal `url` it goes to the `catch (Exception exception)` block, which breaks the original logic of this method.

If we keep the current logic of `getURI`, please also update this usage, and also the logging in `PortalImpl#escapeRedirect`.

Or just add the try-catch-log-debug block back to `getURI` to restore its original contract.

Reply:
@dantewang Got it